src/commands/trigger.rs

```diff
@@ -0,0 +1,119 @@
+use std::{path::PathBuf, sync::Arc};
+
+use anyhow::{bail, Context, Result};
+use clap::Args;
+use reqwest::Url;
+use spin_engine::io::FollowComponents;
+use spin_loader::bindle::BindleConnectionInfo;
+use spin_manifest::Application;
+
+use crate::opts::*;
+
+/// Options shared in common by all trigger implementations.
+#[derive(Args, Debug)]
+pub struct TriggerCommonOpts {
+    /// Pass an environment variable (key=value) to all components of the application.
+    #[clap(long = "env", short = 'e', parse(try_from_str = crate::parse_env_var))]
+    pub env: Vec<(String, String)>,
+
+    /// Log directory for the stdout and stderr of components.
+    #[clap(
+            name = APP_LOG_DIR,
+            short = 'L',
+            long = "log-dir",
+            )]
+    pub log: Option<PathBuf>,
+
+    /// Disable Wasmtime cache.
+    #[clap(
+        name = DISABLE_WASMTIME_CACHE,
+        long = "disable-cache",
+        env = DISABLE_WASMTIME_CACHE,
+        conflicts_with = WASMTIME_CACHE_FILE,
+        takes_value = false,
+    )]
+    pub disable_cache: bool,
+
+    /// Wasmtime cache configuration file.
+    #[clap(
+        name = WASMTIME_CACHE_FILE,
+        long = "cache",
+        env = WASMTIME_CACHE_FILE,
+        conflicts_with = DISABLE_WASMTIME_CACHE,
+    )]
+    pub cache: Option<PathBuf>,
+
+    /// Print output for given component(s) to stdout/stderr
+    #[clap(
+        name = FOLLOW_LOG_OPT,
+        long = "follow",
+        multiple_occurrences = true,
+        )]
+    pub follow_components: Vec<String>,
+
+    /// Print all component output to stdout/stderr
+    #[clap(
+        long = "follow-all",
+        conflicts_with = FOLLOW_LOG_OPT,
+        )]
+    pub follow_all_components: bool,
+}
+
+impl TriggerCommonOpts {
+    pub fn wasmtime_config(&self) -> Result<wasmtime::Config> {
+        let mut wasmtime_config = wasmtime::Config::default();
+        if !self.disable_cache {
+            match &self.cache {
+                Some(p) => wasmtime_config.cache_config_load(p)?,
+                None => wasmtime_config.cache_config_load_default()?,
+            };
+        }
+        Ok(wasmtime_config)
+    }
+
+    pub fn follow_components(&self) -> FollowComponents {
+        if self.follow_all_components {
+            FollowComponents::All
+        } else if self.follow_components.is_empty() {
+            FollowComponents::None
+        } else {
+            let followed = self.follow_components.clone().into_iter().collect();
+            FollowComponents::Named(followed)
+        }
+    }
+
+    pub async fn app_from_env(&self) -> Result<Application> {
+        let working_dir = std::env::var("SPIN_WORKING_DIR").context("SPIN_WORKING_DIR")?;
+        let manifest_url = std::env::var("SPIN_MANIFEST_URL").context("SPIN_MANIFEST_URL")?;
+
+        let mut app = if let Some(manifest_file) = manifest_url.strip_prefix("file://") {
```

Comment:
It might be useful to document the permitted/expected formats - I was finding this a bit hard to read.  Also possibly worth pulling it out into some sort of "parse and load app" function?

Reply:
I'd like to discuss whether this is even a good idea. It might make more sense to reuse/forward loader args from `spin up`, its just very awkward without more refactoring (which I'm working on).

Longer-term we should think about whether/how we want to split up the "application resolution" (fetching all modules, assets) and "execution" phases of `spin up`, which starts getting into production deployment questions.